96769015_Vikram/Adding.cu

```diff
@@ -0,0 +1,65 @@
+#include<iostream>
+using namespace std;
+
+__global__ void Add(float *array1, float *array2, float *out,unsigned int *i)
+{
+	int id = blockIdx.x * blockDim.x + threadIdx.x + *i;
+	out[id] = array1[id] + array2[id];
+	
+}
+
+int main()
+{
+	unsigned int array_Size, bigBlock = 1024*1024;
+		
+	cout << "Enter the size of Array: ";
+	cin >> array_Size;
+
+	float h_array1[array_Size], h_array2[array_Size], h_out[array_Size];
+	
+	int array_Bytes = array_Size * sizeof(float);
+
+	for(int i=0;i<array_Size;i++)
+		h_array1[i] = i;
+
+	for(int i=1,j=0;j<array_Size;j++){
+		h_array2[j] = i;
+		
+	}
+
+	float *d_array1, *d_array2, *d_out;
+	unsigned int *d_i;
+	
+	cudaMalloc((void**)&d_array1, array_Bytes);
+	cudaMalloc((void**)&d_array2,array_Bytes);
+	cudaMalloc((void**)&d_out,array_Bytes);
+	cudaMalloc((void**)&d_i,sizeof(int));
+	
+	cudaMemcpy(d_array1, h_array1, array_Bytes, cudaMemcpyHostToDevice);
+	cudaMemcpy(d_array2, h_array2, array_Bytes, cudaMemcpyHostToDevice);
+	
+	if(array_Size > bigBlock)
+	for(unsigned int *i=0;*i>=array_Size + bigBlock;*i+=bigBlock)
+	{
+		cudaMemcpy(d_i, i, sizeof(int), cudaMemcpyHostToDevice);
+		Add<<<1024,1024>>>(d_array1, d_array2, d_out, d_i);
+	}
```

Comment:
There is no restriction on the number of blocks that you can launch. Only restriction is the number of threads per block, which is 1024. You can launch however many blocks you want and so you can span the entire array in just one kernel call. So this if condition is not required.

Reply:
but still i cant run 30M numbers, I think I'm exceeding the array size Limit.

Reply:
Yeah, that's true, maximum array size that can be used is less than a million.